### PR TITLE
Refactor install 

### PR DESCRIPTION
**Pull request by vid**
This PR uses composer and repos directly. It robustly checks for existing mw installs, and does a fresh install if none is found. A few test are also added.

### CODE REVIEW

**Comment by Phanoix** [comment]
I double checked, [OpenID_Connect](https://www.mediawiki.org/wiki/Special:MyLanguage/Extension:OpenID_Connect) along with its dependency on [PluggableAuth](https://www.mediawiki.org/wiki/Special:MyLanguage/Extension:PluggableAuth) is the only auth extension we'll need to keep around, the MW-OAuth2Client one isn't used anywhere right now

File: site/config-gcpedia.php [hunk 1]
```diff
@@ -199,6 +199,10 @@
 wfLoadExtension("SidebarTitleText");
 
 wfLoadExtension("TimedMediaHandler");
+
+wfLoadExtension("PluggableAuth");
```
Comment (Phanoix) [inline site/config-gcpedia.php:203]: openID is only used in the external wiki

**Comment by Phanoix** [comment]
probably in a separate PR after this is merged we'll have to also get the openID config working from env.

**Comment by vid** [comment]
<insert Monday meme> Fixed now.

**Comment by Phanoix** [comment]
Last thing, something here seems to have broken the init for the PR instances, but that can be fixed separately since this PR's pretty big and has been open for quite a while now.

**Review by Phanoix — APPROVED** [review]

**Pull request merged**